openslides_backend/action/actions/user/forget_password.py

```diff
@@ -0,0 +1,136 @@
+from collections import defaultdict
+from smtplib import (
+    SMTPAuthenticationError,
+    SMTPDataError,
+    SMTPRecipientsRefused,
+    SMTPSenderRefused,
+    SMTPServerDisconnected,
+)
+from ssl import SSLCertVerificationError
+from time import time
+from typing import Any, Dict
+
+from ....models.models import User
+from ....shared.exceptions import ActionException
+from ....shared.filters import FilterOperator
+from ...generics.update import UpdateAction
+from ...mixins.send_email_mixin import EmailMixin, EmailSettings
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ...util.typing import ActionData
+
+PW_FORGET_EMAIL_TEMPLATE = """You receive this email, because you have requested a new password for your OpenSlides-account.
+
+Please open the following link and choose a new password:
+/login/forget-password-confirm?user_id={user_id}&token={token}
+
+For completeness your username: {username}"""
+PW_FORGET_EMAIL_SUBJECT = "Reset your OpenSlides password"
+
+
+class format_dict(defaultdict):
+    def __missing__(self, key: str) -> str:
+        return f"'{key}'"
+
+
+@register_action("user.forget_password")
+class UserForgetPassword(EmailMixin, UpdateAction):
+    """
+    Action to send forget password mail(s).
+    """
+
+    model = User()
+    schema = DefaultSchema(User()).get_default_schema(
+        title="user forget password schema",
+        additional_required_fields={"email": {"type": "string"}},
+    )
+    skip_archived_meeting_check = True
+
+    def get_updated_instances(self, action_data: ActionData) -> ActionData:
+        for instance in action_data:
+            email = instance.pop("email")
+
+            # check emailif valid email adress
+            if not self.check_email(EmailSettings.default_from_email):
+                raise ActionException(
+                    f"email '{EmailSettings.default_from_email}' is not a valid sender email address."
+                )
+            if not self.check_email(email):
+                raise ActionException(f"'{email}' is not a valid email adress.")
+
+            # search for users with email
+            filter_ = FilterOperator("email", "=", email)
+            results = self.datastore.filter(
+                self.model.collection, filter_, ["id", "username"]
+            )
+
+            # try to send the mail.
+            try:
```

Comment:
With the current code, the email will be sent to all users until an error happens. After the error, the method exits. I think we should unify this so all users are tried, regardless if an error happens or not. Afterwards, if any error happened, it should be returned in the action results instead of throwing an exception.

Reply:
If there is a general problem with sending mails, it should stop. 
The exceptions SMTPAuthenticationError, SMTPSenderRefused, SSLCertVerificationError are general exception.
The sending email mechanism is broken. So I stop sending mails.
The SMTPRecipientsRefused, SMTPServerDisconnected, SMTPDataError exception don't stop sending emails.

Reply:
Ah, ok, I see.